ci/cibw_before_all_windows.sh

```diff
@@ -23,9 +24,25 @@ if [[ "$ARCH" == "ARM64" ]]; then
     export LINK="/LIBPATH:$ZLIB_ROOT/lib"
     export HDF5_VSVERSION="17-arm64"
 elif [[ "$ARCH" == "AMD64" ]]; then
-    # NuGet for Windows x64
-    nuget install zlib-msvc-x64 -ExcludeVersion -OutputDirectory "$PROJECT_PATH"
-    ZLIB_ROOT="$PROJECT_PATH/zlib-msvc-x64/build/native"
+    # Build zlib from source for Windows AMD64
+    mkdir zlib_win_x64
+    curl -sLO https://github.com/madler/zlib/archive/refs/tags/v$ZLIB_VERSION.tar.gz
```

Comment:
For Mac, we download a tarball from `https://zlib.net/fossils/zlib-$ZLIB_VERSION.tar.gz`. Does the archive from Github give us anything extra? If not, could we use the same URL on Windows.